yt_dlp/extractor/moviepilot.py

```diff
@@ -0,0 +1,121 @@
+            'title': 'Queen & Slim',  # test escaping
+            'ext': 'mp4',
+            'thumbnail': 'https://s2.dmcdn.net/v/SbUM71WtomSjVmI_q/x720',
+            'timestamp': 1571838685,
+            'description': 'md5:73058bcd030aa12d991e4280d65fbebe',
+            # coming from Dailymotion
+            'uploader': 'Moviepilot',
+            'like_count': int,
+            'view_count': int,
+            'uploader_id': 'x6nd9k',
+            'upload_date': '20191023',
+            'duration': 138,
+            'age_limit': 0,
+            'tags': ['Movie', 'Verleih', 'Neue Trailer'],
+        },
+    }, {
+        'url': 'https://www.moviepilot.de/movies/der-geiger-von-florenz/trailer',
+        'info_dict': {
+            'id': 'der-geiger-von-florenz',
+            'title': 'Der Geiger von Florenz',
+            'ext': 'mp4',
+        },
+        'skip': 'No trailer for this movie.',
+    }, {
+        'url': 'https://www.moviepilot.de/movies/muellers-buero/',
+        'info_dict': {
+            'id': 'x7xcw1i',
+            # 'id': 'muellers-buero',
+            'title': 'Müllers Büro',  # test umlauts
+            'ext': 'mp4',
+            'description': 'md5:57501251c05cdc61ca314b7633e0312e',
+            'timestamp': 1287584475,
+            'age_limit': 0,
+            'duration': 82,
+            'upload_date': '20101020',
+            'thumbnail': 'https://s1.dmcdn.net/v/SaMes1WfAm1d6maq_/x720',
+            'uploader': 'Moviepilot',
+            'like_count': int,
+            'view_count': int,
+            'tags': ['Alle Trailer', 'Movie', 'Verleih'],
+            'uploader_id': 'x6nd9k',
+        },
+    }]
+
+    def _get_property(self, name, webpage, fatal=True):
+        return self._search_regex(f'meta itemprop="{name}" content="(.*?)"', webpage, name, fatal=fatal)
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        # normalize url
+        url = f"https://www.moviepilot.de/movies/{video_id}/trailer"
+
+        webpage = self._download_webpage(url, video_id)
+
+        return {'_type': 'url_transparent',
+                'ie_key': 'Dailymotion',
+                'id': video_id,
+                # Currently, the movie title is extracted, e.g. "Interstellar" or "Müllers Büro"
+                # The actual trailer title is: "Interstellar - Trailer 2 (Deutsch) HD" or
+                # "MÃ¼llers BÃ¼ro - Trailer (Deutsch)", umlauts are really broken here.
+                # I prefer to keep the actual movie title or is there an extra metadata field for that?
+                'title': self._og_search_title(webpage),
+                'url': self._get_property("embedURL", webpage),
+                'thumbnail': self._get_property("thumbnailURL", webpage, fatal=False),
+                # The duration is something like "P00H02M20S" but parse_duration cannot parse this
+                # should I write an extra function for that or should parse_duration be able to parse this?
```

Comment:
You should ask for clarifications on github, not put as comment in code 😐 

To answer this, use regex to either convert this into a parsable format or even directly convert to seconds

Reply:
I have copied parts of the code of `parse_duration` now. Is it necessary to convert to float here (also taken from `parse_duration`)?